app/src/main/java/com/github/wnder/NavigationActivity.java

```diff
@@ -113,21 +113,42 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
         //if it's not the good request, return
         if(requestCode == REQUEST_POSITION_CODE){
             //permission to get the location
-            for(int i = 0; i < permissions.length; ++i){
-                if(permissions[i].equals(Manifest.permission.ACCESS_FINE_LOCATION) && !(grantResults[i] == PackageManager.PERMISSION_GRANTED)){
-                    AlertBuilder.okAlert(getString(R.string.gps_access_title), getString(R.string.gps_access_body), this).show();
-                }
-                else{
-                    LocationManager LocMan = (LocationManager)getSystemService(Context.LOCATION_SERVICE);
-                    if(!LocMan.isProviderEnabled(LocationManager.GPS_PROVIDER)){
-                        AlertBuilder.okAlert(getString(R.string.gps_disabled_title), getString(R.string.gps_disabled_body), this).show();
-                    }
-                    LocMan.requestLocationUpdates(LocationManager.GPS_PROVIDER, 100, 1, location -> {
-                        //Nothing to do in case of location change, the request is being done when necessary with getLastKnownLocation
-                    });
-                }
+            boolean permGranted = checkIfGPSPermissionHasBeenGranted(permissions, grantResults);
+
+            if(permGranted) {
+                checkIfGPSIsActivated();
+            }
```

Comment:
You can write directly the function in the "if", I don't know if it's a better way of doing it but I find it cooler

Reply:
+1 cool for the app